src/Runtime/PyOMCompileExecutionSession.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------ PyOMCompileExecutionSession.hpp - PyOMCompileExecutionSession
+// Declaration -------===//
+//
+//
+// =============================================================================
+//
+// This file contains declaration of PyOMCompileExecutionSession class, which
+// helps python programs to compile and run binary model libraries.
+//
+//===----------------------------------------------------------------------===//
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+
+#include "ExecutionSession.hpp"
+#include "OnnxMlirCompiler.h"
+
+namespace onnx_mlir {
+
+class PyOMCompileExecutionSession : public onnx_mlir::ExecutionSession {
+public:
+  PyOMCompileExecutionSession(std::string inputFileName,
+      std::string sharedLibPath, std::string flags,
+      bool defaultEntryPoint = true);
+  std::string pyGetCompiledFileName();
+  std::string pyGetErrorMessage();
+  int64_t pyGetCompiledResult();
+  std::vector<std::string> pyQueryEntryPoints();
+  void pySetEntryPoint(std::string entryPointName);
+  std::vector<py::array> pyRun(const std::vector<py::array> &inputsPyArray);
+  std::string pyInputSignature();
+  std::string pyOutputSignature();
+
+private:
```

Comment:
Would we not have less code replication if this combined Compile and execution did not inherit from either, but has inside one object `PyOMCompileSession` and one `ExecutionSession` and then we just forward the functions to the right object? Ideally there should be no new code here, right?

Reply:
> Would we not have less code replication if this combined Compile and execution did not inherit from either, but has inside one object `PyOMCompileSession` and one `ExecutionSession` and then we just forward the functions to the right object? Ideally there should be no new code here, right?

Yes. There should be no new code.

Reply:
Cool, hopefully a quick refactor and we can declare success :-)

Reply:
Done.